jquery.xblock.js

```diff
@@ -164,33 +164,62 @@
                 },
                 listenTo: function(name, callback) {
                     $this.dispatcher.bind(name, callback);
+                },
+                children: function(element) {
+                    return $(element).data('xblock-children');
+                },
+                childMap: function(block, childName) {
+                    var children = this.children(block);
+                    for (var i = 0; i < children.length; i++) {
+                        var child = children[i];
+                        if (child.name == childName) {
+                            return child;
+                        }
+                    }
                 }
             };
         },
 
-        jsInit: function(options, root) {
+        initializeXBlocks: function(options, root) {
+            // Find and initialize any XBlocks that are descendants of 'root', and their descendants.
             var $this = this;
+            return $(root).highestDescendants('.xblock:not(.xblock-initialized)').map(function(idx, blockDOM) {
```

Comment:
@e-kolpakov If 'root' here is an XBlock, then you're right. But the way this code is used, 'root' is initially an outer container DIV that contains one or more XBlocks. So root is not initially an XBlock, though it is when this function is called recursively.